### PR TITLE
Updated Segment integration for new Appcues SDK

### PR DESCRIPTION
**Pull request by gamache**
This PR updates the Appcues Segment JS integration:

* support for `Appcues.track`, `Appcues.page`, `Appcues.screen`
* soft failure when `window.Appcues` is not available (this very occasionally happens if the download fails)

### CODE REVIEW

**Comment by hijonathan** [comment]
To make this easier on the Segment folks and improve the chances of this getting merged in quickly, could we instead add these new methods to the old SDK to avoid having to make that check?

File: lib/index.js [hunk 1]
```diff
@@ -63,13 +63,47 @@ Appcues.prototype.load = function(callback) {
...
+ * http://appcues.com/docs#page
  *
  * @api public
+ * @param {Page} page
  */
 
-Appcues.prototype.page = function() {
-  window.Appcues.start();
+Appcues.prototype.page = function(page) {
+  if (!this.loaded()) {
```
Comment (hijonathan) [inline lib/index.js]: I don't think you actually need to do this. Also, I highly doubt they'll allow `console.log` in there.

**Comment by gamache** [comment]
:+1: on adding stubs to the old SDK.

File: lib/index.js [hunk 2]
```diff
@@ -63,13 +63,47 @@ Appcues.prototype.load = function(callback) {
...
+/**
+ * Screen.
+ *
+ * http://appcues.com/docs#screen
+ *
+ * @api public
+ * @param {Screen} screen
+ */
+
+Appcues.prototype.screen = function(screen) {
```
Comment (rayd) [inline lib/index.js]: What is `screen`?

Comment (gamache) [inline lib/index.js]: It's like `page`, basically. https://segment.com/docs/spec/screen/

Comment (hijonathan) [inline lib/index.js]: Looks like that's mobile-specific and not actually a method in their JS SDK.

File: lib/index.js [hunk 3]
```diff
@@ -82,5 +116,33 @@ Appcues.prototype.page = function() {
  */
 
 Appcues.prototype.identify = function(identify) {
-  window.Appcues.identify(identify.userId(), identify.traits());
+  if (!this.loaded()) {
+    console.log('Appcues is not loaded');
+    return;
+  }
+
+  if (window.Appcues.identify) {
```
Comment (rayd) [inline lib/index.js]: We might want to just check that `window.Appcues` exists, so we catch that weird issue that Zetland saw once or maybe twice where `window.Appcues` didn't exist.

File: lib/index.js [hunk 4]
```diff
@@ -63,13 +63,47 @@ Appcues.prototype.load = function(callback) {
...
-Appcues.prototype.page = function() {
-  window.Appcues.start();
+Appcues.prototype.page = function(page) {
+  if (!this.loaded()) {
+    console.log('Appcues is not loaded');
+    return;
+  }
+
+  if (window.Appcues.page) {
+    window.Appcues.page(page.name(), page.properties());
```
Comment (rayd) [inline lib/index.js]: Should we also include the page category which can potentially be passed into the analytics.js `page` call? https://segment.com/docs/libraries/analytics.js/#page

**Comment by rayd** [comment]
This looks good - the only thing I'd say is that we could shim out the Appcues interface in case something goes wrong. Intercom does something along those lines in their integration: https://github.com/segment-integrations/analytics.js-integration-intercom/blob/master/lib/index.js#L34

Might be worth it to be a bit defensive in here, especially while we can get some changes in.

**Pull request merged**